### PR TITLE
Force sending of content type to gcs

### PR DESCRIPTION
**Pull request by digitalsadhu**
Content type was not being detected when files were uploaded.

## Issue
See: #44 

## Status
**IN DEVELOPMENT**

## Description
Intention is to add a full integration test for this in the https://github.com/asset-pipe/integration repo.

For now I have verified (by eye) that the ContentType header is being sent. Before this fix it was not being sent. 

## Todos
- [x] Tests (in integration repo)
- [x] Verify this fix actually solves the issue we have

## Deploy Notes
Once fixed, we will need to update and redeploy the asset pipe server. No other changes should be required.

## Related PRs
* None

### CODE REVIEW

File: lib/sink.js [hunk 1]
```diff
@@ -40,7 +41,17 @@ class WriteStream extends stream.PassThrough {
         const parser =
             type === 'json' ? JSONStream.parse('*') : new stream.PassThrough();
 
-        const gcsStream = gcsFile.createWriteStream(options);
+        const gcsStream = gcsFile.createWriteStream(
+            Object.assign(
```
Comment (SimenB) [inline lib/sink.js]: object spread?

File: lib/sink.js [hunk 2]
```diff
@@ -40,7 +41,17 @@ class WriteStream extends stream.PassThrough {
...
+                {},
+                {
+                    metadata: {
+                        contentType: extname(type)[0] && extname(type)[0].mime,
```
Comment (SimenB) [inline lib/sink.js]: should probably deep-merge `metadata`

File: lib/sink.js (hunk 1 above)
Comment (digitalsadhu) [reply lib/sink.js]: Started with that but eslint was being a knob about it so dropped it. Probably missing a config flag of some kind.

Comment (SimenB) [reply lib/sink.js]:
```json
  "parserOptions": {
    "ecmaVersion": 2017
  },
```

Comment (digitalsadhu) [reply lib/sink.js]: looks about right ;)

File: lib/sink.js [hunk 3]
```diff
@@ -8,6 +8,7 @@ const isString = require('lodash.isstring');
 const common = require('asset-pipe-common');
 const stream = require('readable-stream');
 const assert = require('assert');
+const extname = require('ext-name');
```
Comment (SimenB) [inline lib/sink.js]: what about https://www.npmjs.com/package/mime-types instead?

Comment (digitalsadhu) [reply lib/sink.js]: yea that looks nicer

File: lib/sink.js (hunk 1 above)
Comment (digitalsadhu) [reply lib/sink.js]: done

File: lib/sink.js (hunk 2 above)
Comment (digitalsadhu) [reply lib/sink.js]: done

File: lib/sink.js (hunk 3 above)
Comment (digitalsadhu) [reply lib/sink.js]: done

**Comment by digitalsadhu** [comment]
Can I get a whoop whoop?

**Pull request merged**